fastlane/Fastfile

```diff
@@ -73,29 +73,36 @@ platform :android do
   desc "Upload and close a release"
   lane :deploy do |options|
     version = android_get_version_name(gradle_file: gradle_file_path)
-    is_snapshot_version = version.end_with?("-SNAPSHOT")
-    puts "Deploying #{version}"
-    gradle(
-      tasks: [
-          "androidSourcesJar", "androidJavadocJar", "publish", "closeAndReleaseRepository"
-      ],
-      properties: {
-        "signing.keyId" => ENV['GPG_SIGNING_KEY_ID_NEW'],
-        "signing.password" => ENV['GPG_SIGNING_KEY_PW_NEW'],
-        "signing.secretKeyRingFile" => "/home/circleci/.gnupg/secring.gpg",
-        "mavenCentralUsername" => ENV['SONATYPE_NEXUS_TOKEN_USERNAME'],
-        "mavenCentralPassword" => ENV['SONATYPE_NEXUS_TOKEN_PASSWORD'],
-        "RELEASE_SIGNING_ENABLED" => true,
-      }
-    )
-    if !is_snapshot_version then github_release(version: version) end
+  	is_snapshot_version = version.end_with?("-SNAPSHOT")
+  	gradleProperties = {
+  	  "signing.keyId" => ENV['GPG_SIGNING_KEY_ID_NEW'],
+  		"signing.password" => ENV['GPG_SIGNING_KEY_PW_NEW'],
+  		"signing.secretKeyRingFile" => "/home/circleci/.gnupg/secring.gpg",
+  		"mavenCentralUsername" => ENV['SONATYPE_NEXUS_TOKEN_USERNAME'],
+  		"mavenCentralPassword" => ENV['SONATYPE_NEXUS_TOKEN_PASSWORD'],
+  		"RELEASE_SIGNING_ENABLED" => true
+  	}
+  	puts "Deploying #{version}"
+  	gradle(
+  		tasks: [
+  			"androidSourcesJar", "androidJavadocJar", "publish --no-daemon --no-parallel"
+  		],
+  		properties: gradleProperties
+  	)
+  	gradle(
+  		tasks: [
+  			"closeAndReleaseRepository"
+  		],
+  		properties: gradleProperties
+  	)
+  	if !is_snapshot_version then github_release(version: version) end
   end
 
   desc "Upload a snapshot release"
   lane :deploy_snapshot do |options|
     version = android_get_version_name(gradle_file: gradle_file_path)
     is_snapshot_version = version.end_with?("-SNAPSHOT")
```

Comment:
we're repeating this code in a couple of places, let's extract it into

```ruby
def is_snapshot_version?(version_name)
    version.end_with?("-SNAPSHOT")
end

# usage
version = android_get_version_name(gradle_file: gradle_file_path)
deploy if !is_snapshot_version?(version)
```

the `?` in the method name isn't necessary but it's the convention for boolean methods in ruby

Reply:
resident 💎  expert!!! thank you!